http-server/src/Main.gren

```diff
@@ -0,0 +1,75 @@
+module Main exposing (main)
+
+import Node
+import Bytes exposing (Bytes)
+import Bytes.Encode as BE
+import Node.Program as Program exposing (Program)
+import Node.Http as Http
+
+
+main : Program Model Msg
+main =
+    Program.define
+        { init = init
+        , update = update
+        , subscriptions = subscriptions
+        }
+
+
+type alias Model =
+    { serverKey : Http.ServerKey
+    }
+
+
+type Msg
+    = HandleRequest Http.Request
+    {-
+        type alias Request =
+            { headers : Dict String String
+            , body : Bytes
+            , url : Url
+            , method : Method
+            }
+        
+        type Method
+            = GET
+            | HEAD
+            | POST
+            | PUT
+            | DELETE
+            | CONNECT
+            | TRACE
+            | PATCH
+    -}
+
+
+init : Program.AppInitTask { model : Model, command : Cmd Msg }
+init =
+    Program.await Http.createServer <| \serverKey ->
+        Program.startProgram
+            { model =
+                { serverKey = serverKey
+                }
+            , command =
+                Cmd.none
+            }
+
+
+update : Msg -> Model -> { model : Model, command : Cmd Msg }
+update msg model =
+    case msg of
+        HandleRequest req ->
+            { model = model
+            , command = Http.respondWith model.serverKey
```

Comment:
It might make more sense for `Http.respondWith` to take `req` as the first argument, instead of `serverKey`. `req` itself is a type of permission as you had to have the key in order to create the subscription, and it becomes explicit what you're responding to.

Reply:
Makes sense. I'm thinking `onRequest` doesn't need the key either?